shared/domain/src/main/java/com/ivy/domain/usecase/account/AccountBalanceUseCase.kt

```diff
@@ -23,7 +23,13 @@ class AccountBalanceUseCase @Inject constructor(
         account: AccountId,
         outCurrency: AssetCode
     ): ExchangedAccountBalance {
-        TODO("Not implemented")
+        val balance = calculate(account)
+        val exchangeResult = exchangeUseCase.convert(values = balance, to = outCurrency)
+        return if(balance.keys.size == exchangeResult.exchangeErrors.size || balance.values.sumOf { it.value } == 0.0) {
```

Comment:
I don't understand this `if`. What's its purpose?

Reply:
I was unsure about that. The docs said to return no balance if the balance is zero or if the exchange to outCurrency failed for all assets. So, if the first case occurs, the sum of the values should be 0.0. If the second occurs, the exchangeErrors.size should be equal to the size of all asset codes, right?

Reply:
> If the second occurs, the exchangeErrors.size should be equal to the size of all asset codes, right?

E.g. 2 assets and 2 errors doesn't mean that we should return empty. IMO, this if is unnecessary. The user of the API should check for themselves if there are errors and how to handle them

Reply:
The check should be:

if balances.isEmpty => balance = Option.None

Reply:
errors stay as they are

Reply:
Got it. Thanks!